### DISCUSSION TITLE
FindMultiModalNeighbors on integrated 10X multiome datasets

### DISCUSSION THREAD

**Discussion by JoreVW** [satijalab/seurat#6060] [Q&A]
Dear Seurat/Signac authors,

I'm having trouble with the final step of my multiome data, being the FindMultiModalNeighbors.

>integrated_SCT<- FindMultiModalNeighbors(
+   object = integrated_SCT,
+   reduction.list = list("umap", "lsi"), 
+   dims.list = list(1:15, 2:15),
+   modality.weight.name = "RNA.weight.name",
+   verbose = TRUE)
Calculating cell-specific modality weights
Error in Embeddings(object = object, reduction = r)[, dims.list[[r]]] : 
  subscript out of bounds

I have 2 10X ATAC + GEX datasets, after filtering I merged and integrated based on the RNA assay using CCA. I merged, preformed SCT and found anchors as all described in the vignettes. My map looks like expected, nice integration. 

>features <- SelectIntegrationFeatures(object.list = merged_SCT)
> merged_SCT <- PrepSCTIntegration(object.list = merged_SCT, anchor.features = features)
> anchors_SCT <- FindIntegrationAnchors(object.list = merged_SCT, dims = 1:30, anchor.features = features, normalization.method = "SCT", reduction = "cca")
> integrated_SCT <- IntegrateData(anchorset = anchors_SCT, normalization.method = "SCT", dims = 1:30)
> integrated_SCT<- RunPCA(integrated_SCT)
> integrated_SCT<- RunUMAP(integrated_SCT, dims = 1:15)
> integrated_SCT<- FindNeighbors(integrated_SCT, reduction = "pca", dims = 1:15)
> integrated_SCT<- FindClusters(integrated_SCT, resolution = 0.03)

Next I switched to ATAC assay ("peaks", MACS2 peak calling) and followed the vignette again to integrate my ATAC data based on the RNA-anchors.

> DefaultAssay(integrated_SCT) <- "peaks"
> integrated_SCT <- FindTopFeatures(integrated_SCT, min.cutoff = 10)
> integrated_SCT <- RunTFIDF(integrated_SCT)
> integrated_SCT <- RunSVD(integrated_SCT)
> integrated_SCT <- RunUMAP(integrated_SCT, reduction = "lsi", dims = 2:30)

> integrated_SCT <- IntegrateEmbeddings(
  anchorset = anchors_SCT,
  reductions = integrated_SCT[["lsi"]],
  new.reduction.name = "integrated_lsi",
  dims.to.integrate = 1:30)

Again, comparing the umap's before and after integration, all looks fine as as expected integration via the RNA anchors work really nice.

However, when I want to finalise the vignette and want the Joint Umap, I immediately get the error above when running FindMultiModalNeighbors. I already changed dimensions of k-associated parameters but still the same error.

I could use some help here what to check to fix this error?

Thank you very much, and again thank you as wel for all the Seurat/Signac software. It is a great package and extremely user friendly!

Kind Regards,

Jore Van Wauwe

**Comment by timoast** [comment]
Here you're asking for the first 15 UMAP dimensions, is that what you intend? And are there 15 UMAP dimensions present in the object? The error message suggests not

**Reply by JoreVW** [reply]
Hi @timoast 

it was a really stupid mistake on my part, it should have been PCA, not UMAP. My concentration was below zero after some intens weeks/months of learning bioinfomatics to filter, integrate and analyse 10X multiome datasets. Great work on Seurat/Signac btw, the tools are amazing!

Thanks for the reaction though!

Best, Jore

**Reply by JoreVW** [reply]
@timoast I also noticed that I should save the Atac integration based on the LSI in a new object, not in the original cause the PCA reduction from the integrated RNA will get lost. So I copy my LSI reduction from my new integrated Atac object, in my original Seurat Object that contains the integrated RNA and the PCA needed to run IntegrateEmbeddings
